simulation-construction-set/falling-brick/src/main/java/us/ihmc/fallingBrick/WavyGroundProfile.java

```diff
@@ -0,0 +1,86 @@
+package us.ihmc.fallingBrick;
+
+import us.ihmc.euclid.geometry.BoundingBox3D;
+import us.ihmc.euclid.tuple3D.Point3D;
+import us.ihmc.euclid.tuple3D.Vector3D;
+import us.ihmc.euclid.tuple3D.interfaces.Point3DBasics;
+import us.ihmc.euclid.tuple3D.interfaces.Vector3DBasics;
+import us.ihmc.jMonkeyEngineToolkit.GroundProfile3D;
+import us.ihmc.jMonkeyEngineToolkit.HeightMapWithNormals;
+
+public class WavyGroundProfile implements GroundProfile3D, HeightMapWithNormals
+{
+   private double xMin = -2.0, xMax = 2.0, yMin = -2.0, yMax = 2.0, zMin = -10.0, zMax = 10.0;
+
+   private BoundingBox3D boundingBox = new BoundingBox3D(new Point3D(xMin, yMin, zMin), new Point3D(xMax, yMax, zMax));
+
+   public WavyGroundProfile()
+   {
+   }
+
+   @Override
+   public double heightAndNormalAt(double x, double y, double z, Vector3D normalToPack)
+   {
+      double heightAt = heightAt(x, y, z);
+      surfaceNormalAt(x, y, heightAt, normalToPack);
+      return heightAt;
+   }
+
+   @Override
+   public double heightAt(double x, double y, double z)
+   {
+      // if in bounds, this is the point where the height of the ground is defined
+      // One can change the profile of the terrain by changing this function
+      if ((x > xMin) && (x < xMax) && (y > yMin) && (y < yMax))
+         return 1.0 * Math.exp(-Math.abs(2.0 * x)) * Math.exp(-Math.abs(2.0 * y)) * Math.sin(2.0 * Math.PI * 0.7 * x);
+      else
+         return 0.0;
+   }
+
+   public void surfaceNormalAt(double x, double y, double z, Vector3DBasics normal)
+   {
+      normal.setX(0.0);
+      normal.setY(0.0);
+      normal.setZ(1.0);
```

Comment:
Normally is obviously here, it should be the gradient of the function defined line 35.
Really it does matter that much, the main thing it messes up is the ground contact model will not apply the right set of stiffness&damping along the right axis, but overall the behavior would not change that much if this was properly expressed.
I guess at the end I'd say that we've got to either fix it or comment about it saying that it is wrong and that it should be the gradient but that it does not matter that much in our example.